File: connected-devices-python workspace/iot-device/apps/labs/module07/CoapClientTestApp.py
'''
Created on Mar 7, 2019

@author: khars
'''
from labs.module07.CoapClientConnector import CoapClientConnector
from labs.common.SensorData import SensorData
from labs.common.DataUtil import DataUtil
from coapthon.messages import request
from labs.common.ConfigUtil import ConfigUtil
from labs.common import ConfigConst

config = ConfigUtil('../../../config/ConnectedDevicesConfig.props')
config.loadConfig()
data = DataUtil()
host = config.getProperty(ConfigConst.ConfigConst.COAP_DEVICE_SECTION, ConfigConst.ConfigConst.HOST_KEY)
port = int(config.getProperty(ConfigConst.ConfigConst.COAP_DEVICE_SECTION, ConfigConst.ConfigConst.PORT_KEY))
path = 'temperature'
sensor = SensorData("Temperature",0.0,20.0)
sensor.addValue(3.0)
print(str(sensor))
coapClient = CoapClientConnector(host, port, path)
coapClient.ping()
json_data = data.SensorDataToJson(sensor)
print("json"+json_data)
#coapClient.get() #Get will respond with NOT_FOUND since data object on server is not initialized
coapClient.post((json_data)) #Post JSON to server
coapClient.get()
sensor.addValue(4.00)
coapClient.put(data.SensorDataToJson(sensor)) #Update resource on the server
coapClient.get()
coapClient.delete() #Delete resource
coapClient.get()
coapClient.stop()

File: connected-devices-java workspace/iot-gateway/src/main/java/schooldomain/studentname/connecteddevices/labs/module07/CoapClientTestApp.java
package schooldomain.studentname.connecteddevices.labs.module07;

import java.io.IOException;

import schooldomain.studentname.connecteddevices.common.DataUtil;
import schooldomain.studentname.connecteddevices.common.SensorData;

/**
 * This application connects to CoAP server and runs several CURD operations
 * 
 * @author khars
 *
 */
public class CoapClientTestApp {
	public static void main(String[] args) throws IOException
	{		
		runTests();
	}
	
	/**
	 * This method runs tests on the CoAP server with GET, POST etc. actions
	 * 
	 * @throws IOException
	 */
	public static void runTests() throws IOException
	{
		System.out.println("Starting client...");
		CoapClientConnector coapClient = new CoapClientConnector("coap://127.0.0.1/temperature");
				
		SensorData data = new SensorData(20.0,0.0,"time","Temperature");
		data.updateValue(4);
		DataUtil dat = new DataUtil();
		System.out.println(data);
		
		//Ping the server
		coapClient.ping();
		
		//Post to the server to initialize data, then get the data back as JSON
		coapClient.get();
		coapClient.post(dat.SensorDataToJson(data));
		coapClient.get();
		
		//Update resource on server
		data.updateValue(5);
		System.out.println(data);
		coapClient.put(dat.SensorDataToJson(data));
		coapClient.get();
		
		//Delete data
		coapClient.delete();
		coapClient.get();
	}
}